lib/galaxy/webapps/galaxy/services/wes.py

```diff
@@ -0,0 +1,1180 @@
+                instance = instance_str == "true"
+
+        return workflow_id, instance
+
+    except ValueError as e:
+        raise exceptions.MessageException(f"Invalid gxworkflow:// URI: {str(e)}")
+    except Exception as e:
+        raise exceptions.MessageException(f"Error parsing gxworkflow:// URI: {str(e)}")
+
+
+def _load_workflow_content(
+    trans: ProvidesUserContext,
+    workflow_attachment: Optional[UploadFile],
+    workflow_url: Optional[str],
+) -> dict[str, Any]:
+    """Load workflow content from attachment or URL.
+
+    Handles three input methods:
+    1. workflow_attachment: Uploaded file
+    2. workflow_url with gxworkflow:// scheme: Load from Galaxy database
+    3. workflow_url with other schemes: Fetch from URL (http, base64, etc.)
+
+    Returns the workflow as a dictionary for normalization.
+
+    Args:
+        trans: Galaxy transaction/context
+        workflow_attachment: Uploaded workflow file
+        workflow_url: URL to fetch workflow from
+
+    Returns:
+        Dictionary representation of the workflow
+
+    Raises:
+        exceptions.MessageException: If workflow cannot be loaded
+    """
+    workflow_content = None
+
+    # Load from attachment file
+    if workflow_attachment:
+        try:
+            # Read the uploaded file content
+            content = workflow_attachment.file.read()
+            workflow_content = content.decode("utf-8")
+        except UnicodeDecodeError:
+            raise exceptions.MessageException("Workflow attachment must be UTF-8 encoded text")
+        except Exception as e:
+            raise exceptions.MessageException(f"Error reading workflow attachment: {str(e)}")
+
+    # Load from URL
+    elif workflow_url:
+        # Handle gxworkflow:// scheme (Galaxy database reference)
+        if workflow_url.startswith("gxworkflow://"):
+            # This will be handled by the WesService that has access to workflows_manager
+            # Return a special marker dict that includes the URI for later processing
+            return {"workflow_uri": workflow_url}
+
+        # Handle other URL schemes (http, https, base64, file, etc.)
+        try:
+            # Fetch workflow content from URL
+            # validate_uri_access is called by stream_url_to_str for security
+            workflow_content = stream_url_to_str(workflow_url, trans.app.file_sources)
+        except exceptions.AuthenticationRequired as e:
+            raise e
+        except Exception as e:
+            raise exceptions.MessageException(f"Error fetching workflow from URL: {str(e)}")
+
+    if workflow_content is None:
+        raise exceptions.RequestParameterMissingException("Either workflow_url or workflow_attachment must be provided")
+
+    assert isinstance(workflow_content, str)
+
+    # Parse as JSON or YAML
+    try:
+        return json.loads(workflow_content)
+    except json.JSONDecodeError:
+        # Try as YAML-like dict string - Galaxy's normalize_workflow_format
```

Comment:
will that fail in a meaningful way ? maybe yaml.safe_loads which handles JSON and YAML ?